src/main/java/com/kuljava/swiatwsi/rawmaterials/Scheduler.java

```diff
@@ -0,0 +1,30 @@
+package com.kuljava.swiatwsi.rawmaterials;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Scheduler {
+
+  @Autowired RawMaterialsRepository repository;
+
+  private Long woodIncrease;
+  private Long clayIncrease;
+  private Long ironIncrease;
+
+  public Scheduler(
+      @Value("${rawmaterials.scheduler.woodincrease}") String woodIncrease,
+      @Value("${rawmaterials.scheduler.clayincrease}") String clayIncrease,
+      @Value("${rawmaterials.scheduler.ironincrease}") String ironIncrease) {
+    this.woodIncrease = Long.parseLong(woodIncrease);
+    this.clayIncrease = Long.parseLong(clayIncrease);
+    this.ironIncrease = Long.parseLong(ironIncrease);
+  }
+
+  @Scheduled(fixedRateString = "${rawmaterials.scheduler.fixedRate}")
```

Comment:
Extract this parameter to the class field

Reply:
Sheduled annotation has a problem with using a field even final as a fixedRate